### PR TITLE
tests for unique ids across units

### PR DESCRIPTION
**Pull request by IssaHanou**
closes GH-201

unique id's across units

### CODE REVIEW

File: back-end/src/sciler/config/configHandler.go [hunk 1]
```diff
@@ -173,6 +175,48 @@ func generateGeneralTimer(timer ReadTimer, timerMap map[string]*Timer) (*Timer,
...
+	return append(errorList, append(deviceErrors, ruleErrors...)...)
+}
+
+// checkDeviceUniqueIDs checks whether the devices do not have any ids in common with the timers in the config
+func checkDeviceUniqueIDs(idList map[string]string, config *WorkingConfig) (map[string]string, []string) {
+	errorList := make([]string, 0)
+	for deviceID := range config.Devices {
+		if value, ok := idList[deviceID]; ok {
+			errorList = append(errorList,
+				fmt.Sprintf("level III - implementation error: checking devive with id %s, but a %s with id %s already exists",
```
Comment (GwennanS) [inline back-end/src/sciler/config/configHandler.go]: typo devive

File: back-end/src/sciler/config/configHandler.go [hunk 2]
```diff
@@ -173,6 +175,48 @@ func generateGeneralTimer(timer ReadTimer, timerMap map[string]*Timer) (*Timer,
...
 
+// checkUniqueIDs checks whether all timers, devices and rules have unique ids compared to each other.
+func checkUniqueIDs(config *WorkingConfig) []string {
+	idList := make(map[string]string, 0) // the value keeps track of type (rule/timer/device) to put in error message
+	errorList := make([]string, 0)
+	for timerID := range config.Timers {
+		idList[timerID] = "timer"
+	}
+	idList, deviceErrors := checkDeviceUniqueIDs(idList, config)
+	_, ruleErrors := checkRuleUniqueIDs(idList, config)
```
Comment (mrlomar) [inline back-end/src/sciler/config/configHandler.go]: timer IDs should also be unique

File: back-end/src/sciler/config/configHandler.go [hunk 3]
```diff
@@ -173,6 +175,48 @@ func generateGeneralTimer(timer ReadTimer, timerMap map[string]*Timer) (*Timer,
 	}
 }
 
+// checkUniqueIDs checks whether all timers, devices and rules have unique ids compared to each other.
+func checkUniqueIDs(config *WorkingConfig) []string {
```
Comment (mrlomar) [inline back-end/src/sciler/config/configHandler.go]: check methods should be in `checkConfig.go` file

File: back-end/src/sciler/config/configHandler.go [hunk 4]
```diff
@@ -173,6 +175,48 @@ func generateGeneralTimer(timer ReadTimer, timerMap map[string]*Timer) (*Timer,
...
+		if value, ok := idList[ruleID]; ok {
+			errorList = append(errorList,
+				fmt.Sprintf("level III - implementation error: checking rule with id %s, but a %s with id %s already exists",
+					ruleID, value, ruleID))
+		} else {
+			idList[ruleID] = "rule"
+		}
+	}
+	return idList, errorList
+}
```
Comment (mrlomar) [inline back-end/src/sciler/config/configHandler.go]: almost looks like code duplication, can this not be some general method, then used 3 times for rules, devices and timers

File: back-end/src/sciler/config/configHandler.go [hunk 5]
```diff
@@ -247,6 +291,7 @@ func generateStatusMap(config *WorkingConfig) map[string][]*Rule {
 
 	for _, rule := range rules {
 		for _, id := range rule.Conditions.GetConditionIDs() {
+
```
Comment (mrlomar) [inline back-end/src/sciler/config/configHandler.go]: why this enter?

File: back-end/src/sciler/config/configHandler.go (hunk 4 above)
Comment (IssaHanou) [reply back-end/src/sciler/config/configHandler.go]: no than you have more than 2 parameters

**Pull request merged**